pyproject.toml

```diff
@@ -1,14 +1,212 @@
+[project]
+name = "astropy"
+dynamic = [
+    "version"
+]
+description = "Astronomy and astrophysics core library"
+readme = "README.rst"
+authors = [
+    { name = "The Astropy Developers", email = "astropy.team@gmail.com" }
+]
+license = { file = "LICENSE.rst" }
```

Comment:
Would be nice to retain the SPDX identifier that was added some time ago.
```
license = { text = "BSD-3-Clause" }
```
Seems more informative. I'm not sure if we can put both in `license`, and I cannot find a clear recommendation  (https://peps.python.org/pep-0621/#license). We can also use setuptools' `license-files` to specify the file.

Reply:
Unfortunately, we cannot do both, per PEP621 (last sentence):

> The table may have one of two keys. The file key has a string value that is a relative file path to the file which contains the license for the project. Tools MUST assume the file’s encoding is UTF-8. The text key has a string value which is the license of the project whose meaning is that of the License field from the [core metadata](https://packaging.python.org/specifications/core-metadata/). __These keys are mutually exclusive, so a tool MUST raise an error if the metadata specifies both keys.__

So I had to use the `tool.setuptools.liscense-files` to specify the actual files, see 608019c

Reply:
Since @saimn 's last comment is addressed, I am going to squash and merge. Thanks, everyone!